src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodValidateAnonymousClass.java

```diff
@@ -0,0 +1,26 @@
+/*
+JavadocMethod
+allowedAnnotations = MyAnnotation, Override
+validateThrows = (default)false
+accessModifiers = (default)public, protected, package, private
+allowMissingParamTags = (default)false
+allowMissingReturnTag = (default)false
+validateInAnonymousClasses = true
+tokens = (default)METHOD_DEF, CTOR_DEF, ANNOTATION_FIELD_DEF, COMPACT_CTOR_DEF
+
+
+*/
+
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocmethod;
+
+public class InputJavadocMethodValidateAnonymousClass {
+
+     Thread ob = new Thread(){
```

Comment:
Please add more examples of anonymous classes usages.
You can copy ideas from diff report.
Let's have anon classes in ~10 different places: in method, in expression, as parameters,....

Reply:
@romani Can you please tell me how can i use the diff  report for the ideas

Reply:
In PR https://github.com/checkstyle/checkstyle/pull/13968

You have report https://github.com/checkstyle/checkstyle/pull/13968#issuecomment-1788620332
It has a lot of differences, anonymous classes in many areas of code.
But in your PR you have only single test. Yes, is is minimal required test, but we should not depend on diff report too much, we should have in our code base set of Inputs that has targeting code in almost all possible positions.

In short: copy code to inputs from diff report to have anonymous class in not only in field of class.